src/network/network.cpp

```diff
@@ -847,21 +885,25 @@ void Network::UpdateClient()
 				status = NETWORK_STATUS_CONNECTED;
 				server_connection.ResetLastPacketTime();
 				Client_Send_AUTH(gConfigNetwork.player_name, "");
-				window_network_status_open("Authenticating...");
+				char str_authenticating[256];
+				format_string(str_authenticating, STR_MULTIPLAYER_AUTHENTICATING, NULL);
+				window_network_status_open(str_authenticating);
 			}
 		}
 	}break;
 	case NETWORK_STATUS_CONNECTED:
 		if (!ProcessConnection(server_connection)) {
-			std::string errormsg = "Disconnected";
-			if (server_connection.last_disconnect_reason) {
-				errormsg += ": ";
-				errormsg += server_connection.last_disconnect_reason;
-			}
 			if (server_connection.authstatus == NETWORK_AUTH_REQUIREPASSWORD) { // Do not show disconnect message window when password window closed/canceled
 				window_network_status_close();
 			} else {
-				window_network_status_open(errormsg.c_str());
+				char str_disconnected[256], str_conditional_buffer[256];
+				str_conditional_buffer[0] = '\0';
+				if (server_connection.getLastDisconnectReason()) {
```

Comment:
Wouldn't this be better localisable by having a `Disconnected: {STRING}` as well as `DISCONNECTED` as strings? That way you don't need to make an assumption on the order of the message.